src/Containers/Microsoft.NET.Build.Containers/ImageIndexGenerator.cs

```diff
@@ -2,99 +2,120 @@
-using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
 using Microsoft.NET.Build.Containers.Resources;
-using Microsoft.NET.Build.Containers.Tasks;
 
 namespace Microsoft.NET.Build.Containers;
 
-internal readonly struct ImageInfo
-{
-    internal string Config { get; init; }
-    internal string ManifestDigest { get; init; }
-    internal string Manifest { get; init; }
-    internal string ManifestMediaType { get; init; }
-
-    public override string ToString() => ManifestDigest;
-}
-
 internal static class ImageIndexGenerator
 {
     /// <summary>
     /// Generates an image index from the given images.
     /// </summary>
-    /// <param name="imageInfos"></param>
+    /// <param name="images"></param>
     /// <returns>Returns json string of image index and image index mediaType.</returns>
     /// <exception cref="ArgumentException"></exception>
     /// <exception cref="NotSupportedException"></exception>
-    internal static (string, string) GenerateImageIndex(ImageInfo[] imageInfos)
+    internal static (string, string) GenerateImageIndex(BuiltImage[] images)
     {
-        if (imageInfos.Length == 0)
+        if (images.Length == 0)
         {
             throw new ArgumentException(string.Format(Strings.ImagesEmpty));
         }
 
-        string manifestMediaType = imageInfos[0].ManifestMediaType;
+        string manifestMediaType = images[0].ManifestMediaType;
 
-        if (!imageInfos.All(image => string.Equals(image.ManifestMediaType, manifestMediaType, StringComparison.OrdinalIgnoreCase)))
+        if (!images.All(image => string.Equals(image.ManifestMediaType, manifestMediaType, StringComparison.OrdinalIgnoreCase)))
         {
             throw new ArgumentException(Strings.MixedMediaTypes);
         }
 
         if (manifestMediaType == SchemaTypes.DockerManifestV2)
         {
-            return GenerateImageIndex(imageInfos, SchemaTypes.DockerManifestV2, SchemaTypes.DockerManifestListV2);
+            return (GenerateImageIndex(images, SchemaTypes.DockerManifestV2, SchemaTypes.DockerManifestListV2), SchemaTypes.DockerManifestListV2);
         }
         else if (manifestMediaType == SchemaTypes.OciManifestV1)
         {
-            return GenerateImageIndex(imageInfos, SchemaTypes.OciManifestV1, SchemaTypes.OciImageIndexV1);
+            return (GenerateImageIndex(images, SchemaTypes.OciManifestV1, SchemaTypes.OciImageIndexV1), SchemaTypes.OciImageIndexV1);
         }
         else
         {
             throw new NotSupportedException(string.Format(Strings.UnsupportedMediaType, manifestMediaType));
         }
     }
 
-    private static (string, string) GenerateImageIndex(ImageInfo[] images, string manifestMediaType, string imageIndexMediaType)
+    internal static string GenerateImageIndex(BuiltImage[] images, string manifestMediaType, string imageIndexMediaType)
     {
+        if (images.Length == 0)
+        {
+            throw new ArgumentException(string.Format(Strings.ImagesEmpty));
```

Comment:
```suggestion
            throw new ArgumentException(Strings.ImagesEmpty);
```

No need for format (same on line 23)

Reply:
thanks! done